app/services/crawlee_bridge.py

```
@@ -216,6 +217,102 @@ async def save_jobs_to_database(
+                stderr=asyncio.subprocess.PIPE,
+                cwd=self.project_root
+            )
+            
+            try:
+                stdout, stderr = await asyncio.wait_for(
+                    process.communicate(), 
+                    timeout=self.node_timeout
+                )
+            except asyncio.TimeoutError:
+                process.kill()
+                raise CrawleeIntegrationError(f"Multi-site scraping timeout after {self.node_timeout} seconds")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)